### PR TITLE
Switch to micrometer (WIP)

### PR DESCRIPTION
**Pull request by clement26695**
-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by PierreBesson** [comment]
@clement26695 You have put an @Timed on every RestController endpoint but as those are instrumented automatically by Micrometer, I wonder if they are actually needed.

File: generators/server/templates/src/main/java/package/web/rest/AccountResource.java.ejs [hunk 1]
```diff
@@ -23,7 +23,7 @@ import <%=packageName%>.service.UserService;
 import <%=packageName%>.service.dto.UserDTO;
 import <%=packageName%>.web.rest.errors.InternalServerErrorException;
 
-import com.codahale.metrics.annotation.Timed;
+import io.micrometer.core.annotation.Timed;
```
Comment (atomfrede) [inline generators/server/templates/src/main/java/package/web/rest/AccountResource.java.ejs]: I think the imported can be removed (as the methods are not annotated with `@Timed` anymore.

File: generators/server/templates/src/main/java/package/web/rest/AccountResource.java.ejs [hunk 2]
```diff
@@ -88,7 +86,7 @@ public class AccountResource {
     }
 }
 <%_ } else { _%>
-import com.codahale.metrics.annotation.Timed;
+import io.micrometer.core.annotation.Timed;
```
Comment (atomfrede) [inline generators/server/templates/src/main/java/package/web/rest/AccountResource.java.ejs]: I think the imported can be removed (as the methods are not annotated with @Timed anymore.

File: generators/server/templates/src/main/java/package/web/rest/AuthResource.java.ejs [hunk 3]
```diff
@@ -18,7 +18,7 @@
 -%>
 package <%=packageName%>.web.rest;
 
-import com.codahale.metrics.annotation.Timed;
+import io.micrometer.core.annotation.Timed;
```
Comment (atomfrede) [inline generators/server/templates/src/main/java/package/web/rest/AuthResource.java.ejs]: See above :)

File: generators/server/templates/build.gradle.ejs [hunk 4]
```diff
@@ -238,14 +238,6 @@ dependencies {
...
-    compile "io.dropwizard.metrics:metrics-core"
-    <%_ if (['ehcache', 'infinispan'].includes(cacheProvider)) { _%>
-    compile "io.dropwizard.metrics:metrics-jcache"
-    <%_ } _%>
-    compile "io.dropwizard.metrics:metrics-json"
-    compile "io.dropwizard.metrics:metrics-jvm"
-    compile "io.dropwizard.metrics:metrics-servlet"
-    compile "io.dropwizard.metrics:metrics-servlets"
     compile "io.prometheus:simpleclient"
     compile "io.prometheus:simpleclient_dropwizard"
```
Comment (atomfrede) [inline generators/server/templates/build.gradle.ejs]: Are these still required? At least you removed them for maven as far as I can tell from this MR.

**Comment by jdubois** [comment]
This is merged!!!!! Big congrats to @clement26695 !! There are a few bugs to correct, which we have seen together, but this is 99% fine.

**Comment by PierreBesson** [comment]
Congrats @clement26695 ! This is really some impressive work you did touching to almost every part of JHipster.

**Comment by tubbynl** [comment]
Nice work guys 👍 
This implements #7100 , migration-wise; will this be the new default and/or will this be in the next major? :)

**Comment by sazeez** [comment]
@jdubois, I don't see an update to the metrics page in jHipster Registry, and I can confirm metrics is not loading in my local Registry v4.0.6 for v5.8.0 microservices. Can you please confirm?

**Comment by jdubois** [comment]
Uh oh I totally forgot I also needed to upgrade the JHipster Registry :-(
My bad! No worries, the code should be ready, I'll do a patch release early next week.

**Comment by sazeez** [comment]
Thanks, thought I was losing it. :)

**Comment by PierreBesson** [comment]
@sazeez Yes the metrics page is broken as we have moved to Micrometer. This is known and was considered not a priority for now. However I think @clement26695 will have some time the coming weeks to fix it. However he is also working on prometheus dashboards which are more useful for prod as they show metrics over time.

**Comment by PierreBesson** [comment]
@jdubois No need to do a patch release, the work to migrate the registry screen to prometheus format has not been done yet. I guess @sazeez you can open an issue on JHipster/jhipster-registry. Sorry about the bug.

**Comment by jdubois** [comment]
Thanks @PierreBesson - I thought this was done, the page is basically the same as the generated one, so there's not a big work to implement it.

**Comment by clement26695** [comment]
Yes, I forgot to finish this. I just created a PR in the JHipster Registry repository to fix that (jhipster/jhipster-registry#325)

**Comment by gmarziou** [comment]
It would be nice if there was a small migration guide giving some hints on how to migrate exiting apps.
Also a mention about the registry switch to micrometer that implies (unless I"m wrong) that all microservices should be migrated to micrometer all together and we cannot mix with dropwizard.

**Comment by PierreBesson** [comment]
@gmarziou @clement26695 I guess there is a docs PR missing.

**Comment by clement26695** [comment]
I didn't took the time to do it yet. I will try to do it this afternoon or tomorrow

**Pull request merged**